integration-provisioning/src/main/java/org/symphonyoss/integration/provisioning/service/UserService.java

```diff
@@ -129,6 +129,25 @@ private void createNewUser(String sessionToken, String username, String name, St
     }
   }
 
+  /**
+   * Updates user information
+   * @param sessionToken Token to access the User API.
+   * @param user User object
+   * @param name User display name
+   * @param avatar User avatar (Base64 encoded)
+   */
+  private void updateUser(String sessionToken, UserV2 user, String name, String avatar) {
+    UserAttributes userAttributes = createUserAttributes(user.getUsername(), name);
+
+    try {
+      Long userId = user.getId();
+      userApi.v1AdminUserUidUpdatePost(sessionToken, userId, userAttributes);
+      updateUserAvatar(sessionToken, userId, avatar);
+    } catch (ApiException e) {
+      throw new UpdateUserException("Failed to update user avatar", e);
```

Comment:
@dnsymphony The idea is encapsulating all the checked exceptions in runtime exceptions, because inside the provisioning tool the caller shouldn't deal with exceptions, any exception should stop the application execution and reports the execution summary (which integrations were provisioned). 

UpdateUserException is not a subclass of ApiException, is a runtime exception to identify issues during the user update. It wraps the ApiException to know what's the root cause.